.github/workflows/docs-deploy-preview.yaml

```
@@ -0,0 +1,86 @@
+          name: webapp
+          path: ./webapp
+          run-id: ${{ github.event.workflow_run.id }}
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Place web app under subpath
+        run: |
+          mkdir -p ./docs/build/current/client
+          cp -r ./webapp/. ./docs/build/current/client/
+
+      - name: Deploy PR preview to gh-pages
+        uses: peaceiris/actions-gh-pages@v3
```

actionlint (1.7.12)

[error] the runner of "peaceiris/actions-gh-pages@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)